### DISCUSSION TITLE
targets markdown chunks error strangely and fire early when named 'setup'

### DISCUSSION THREAD

**Discussion by KaiAragaki** [ropensci/targets#972] [Help] [answered]
````
```{r}
library(targets)
tar_unscript()
```
````

````
```{targets setup}
tar_target(foo, "bar")
```
Registered S3 method overwritten by 'data.table':
  method           from
  print.data.table
Error in if (record) "enable" else "inhibit" : argument is of length zero
````

Things to note:
1. This does not happen with `r` chunks
2. This only happens with chunks named 'setup' - not  'setu' or 'setups'
3. Including a chunk with this name causes it to fire 'early' - even running the first chunk causes the setup chunk to be run.

If this is a privileged chunk name, it may be nice to have a clearer error.

(Beautiful package by the way. Really one of my favorites of all time.)

**Comment by KaiAragaki** [comment]
Also: Once the error has been met, a simple renaming will not fix it. The R session must be restarted before renaming will work.

**Comment by wlandau** [comment, marked as answer]
I hadn't anticipated that a `{targets}` chunk would also be a setup chunk. Not sure yet how I feel about this as a pattern, but it does technically work now in the development version as of https://github.com/ropensci/targets/commit/32ef529c1c9819c4f1676a67efe28c476e71be6a.

**Reply by KaiAragaki** [reply]
I understand my issue now: I was unaware that chunks named `setup` have a special property and are treated specially. As Alison Hill mentioned, information about this feature is fairly sparse - however, it is mentioned [here](https://r4ds.had.co.nz/r-markdown.html#chunk-name). My original intent was not to make a `targets` chunk a setup chunk, but simply have a `targets` chunk *named* setup. If I had know this previously, I simply would have chosen a different name (say, 'targets-setup').
